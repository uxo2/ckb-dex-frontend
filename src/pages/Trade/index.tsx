import React from 'react'
import { Form, Input, Button, Popover } from 'antd'
import i18n from '../../utils/i18n'
import { useAppState } from '../../contexts/providers'
import {
  TradePage,
  TradeContent,
  TradeForm,
  ConectIconBox
} from './styled'

console.log(useAppState)

export default () => {
  const FormLayoutDemo = () => {
    const [ form ] = Form.useForm()
    const PopverContent = (
      <ul>
        <li>Nothing</li>
      </ul>
    )
    return (
      <div className="formBox">
        <div className="header">
          <span>{i18n.t('trade.pair')}</span>
          <Popover placement="bottomRight" title='' content={ PopverContent } trigger="click">
            <span></span>
            <Button type="text">
              DAI
            </Button>
          </Popover>
          /
          <Popover placement="bottomRight" title='' content={ PopverContent } trigger="click">
            <span></span>
            <Button type="text">
              CKB
            </Button>
          </Popover>
          <Button type="text"></Button>
        </div>
        <Form
          layout="vertical"
          form={form}
          className="form"
          initialValues={{
            layout: "horizontal",
          }}
        >
          <Form.Item>
            <span>You Pay</span>
            <Button className="max-btn" size="small" type="primary">Max</Button>
            <Input placeholder="Enter your Pay" suffix="RMB" type="number" />
          </Form.Item>
          <Form.Item>
            <span>Rate</span>
            <Button className="max-btn" size="small" type="primary">Current</Button>
            <Input placeholder="Rate" suffix="DAI / CKB" type="number" />
          </Form.Item>
          <ConectIconBox>
            <Button shape="circle"></Button>
          </ConectIconBox>
          <Form.Item label="You Receive">
          <Input placeholder="Receive" suffix="DAI" disabled />
          </Form.Item>
          <Form.Item>
            <Button type="primary" style={{ display: "flex", margin: "0 auto" }}>Connect Wallet</Button>
          </Form.Item>
        </Form>
      </div>
    )
  }
  return (
    <TradePage className="Trade">
      <TradeContent>
        <TradeForm>
          <FormLayoutDemo />
        </TradeForm>
      </TradeContent>
    </TradePage>
  )
}
