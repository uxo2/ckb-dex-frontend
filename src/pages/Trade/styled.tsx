import styled from 'styled-components'

export const TradePage = styled.div`
  width: 100%;
  margin: 40px auto 0;
  padding: 0 30px;
  max-width: 1400px;
  .formBox {
    border: 1px solid #e8e8e8;
    border-radius: 10px;
    .header {
      padding: 5px 10px;
      border-bottom: 1px solid #e8e8e8;
    }
    .form {
      padding: 10px;
      .max-btn {
        float: right;
        margin-bottom: 10px;
      }
    }
  }
`
export const TradeForm = styled.div`
  width: 400px;
  height: 600px;
`

export const ConectIconBox = styled.div`
  text-align: center;
`