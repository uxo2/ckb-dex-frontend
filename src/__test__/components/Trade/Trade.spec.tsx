import React, { ReactElement } from 'react'
import renderer from 'react-test-renderer'
import Trade from '../../../pages/Trade'

describe('Header Component render', () => {
  let component: ReactElement

  beforeAll(() => {
    component = <Trade />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
