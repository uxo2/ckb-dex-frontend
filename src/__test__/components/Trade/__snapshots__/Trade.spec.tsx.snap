// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component render shallow renders 1`] = `null`;
